### DISCUSSION TITLE
Renovate adds suffix to the URL in go.mod file, making the URL incorrect

### DISCUSSION THREAD

**Discussion by gjorgi93** [renovatebot/renovate#9678] [Request Help]
After running renovate on my golang packages, on most of the packages the bot added suffix in the URL which shouldn't be there. For example:

It produced `github.com/Azure/azure-sdk-for-go/v53 v53.2.0` instead of `github.com/Azure/azure-sdk-for-go v53.2.0`.
The `/v53` suffix shouldn't be there, thus the `go get -d ./...` command fails with:

```
Command failed: go get -d ./...
go: github.com/Azure/azure-sdk-for-go/v53@v53.2.0: missing github.com/Azure/azure-sdk-for-go/go.mod and .../v53/go.mod at revision v53.2.0
```

Another example:

```
Command failed: go get -d ./...
go: github.com/gofrs/uuid/v4@v4.0.0: missing github.com/gofrs/uuid/go.mod and .../v4/go.mod at revision v4.0.0
```

The `/v4` shouldn't be appended there.

Is there any way how can i remove this suffix in the URL? Thanks

**Comment by rarkins** [comment]
Were these during major *upgrades* or added to packages that you weren't upgrading? Can you create a small public repo to demonstrate?

**Reply by gjorgi93** [reply]
Thanks @rarkins .
Yes, these were during major upgrades.
Yes, demonstration:
https://gitlab.com/dzoonix/renovate-go-packages/-/jobs/1202851823

MR: https://gitlab.com/dzoonix/renovate-go-packages/-/merge_requests/4

**Reply by rarkins** [reply]
This seems to have been added in https://github.com/renovatebot/renovate/pull/8946 by @56KBs 

Hopefully he sees this comment and can advise

**Reply by 56KBs** [reply]
Yeah I would certainly agree this seems likely to have been when this was introduced, there is however a difficulty here - None of these examples look to be using go modules which means to fix this, Renovate would need to be able to identify when a dependency is using Go modules or not.

I don't have any specific knowledge of the logic that Go uses to identify the module system in use by the project so I'd have to do some research to see if this can be resolved somehow

**Reply by rarkins** [reply]
@zharinov would our future GOPROXY use help with this?

**Reply by zharinov** [reply]
Hm, I'm not sure, but I can check on minimal reproduction repo (or fork)

**Reply by rarkins** [reply]
@56KBs would it work to use logic like "if the current version is already >2, and no suffix is present then don't add one"?

**Reply by 56KBs** [reply]
> @56KBs would it work to use logic like "if the current version is already >2, and no suffix is present then don't add one"?

That does sound like a reasonable fix for the logic - It would still leave edge cases for when libraries update to support Go modules, but that would still be a much larger piece of work to undertake

**Comment by tboerger** [comment]
These version suffixes at the end of the import path don't look wrong to me, that's the right format for modules import paths. But your source should also implement these paths.
